### ISSUE TITLE
Swagger : Problems in installing swagger and containers are in 'restarting' state

### ISSUE THREAD

**Issue by vikas027** [vmware/harbor#859]
If you are reporting a problem, please make sure the following information are provided:
1)Version of docker engine and docker-compose
2)Config files of harbor, you can get them by packaging "Deploy/config" directory
3)Log files, you can get them by package the /var/log/harbor/ 

# Environment

- Docker Engine 1.12.1
- Docker Compose 1.8.1
- Python Version 2.7.5
- CentOS 7.2
- Harbor server 0.4.1 (http://192.168.1.88)
- Swagger on 192.168.1.90

# Problem

Followed this [guide](https://github.com/vmware/harbor/blob/master/docs/configure_swagger.md) to test swagger, had to do some tweaks (first three lines below) before `docker-compose build`, otherwise it was throwing errors for `env` files not found; like these  `ERROR: Couldn't find env file: /root/harbor/Deploy/config/ui/env`.

```
# cat templates/ui/env > config/ui/env
# mkdir config/db && cat templates/db/env > config/db/env
# cat templates/jobservice/env > config/jobservice/env
# docker-compose build
# docker-compose up -d
```
 
And, containers `deploy_jobservice_1` and `deploy_registry_1` keeps going in `restarting` state. Probably there is some issue(s) with the variables defined in `env` files.
![2016-10-06_19-16-30](https://cloud.githubusercontent.com/assets/6998650/19145263/f800cf9c-8bf9-11e6-9650-e3965ae95e5a.png)

Logs files are here 
[2016-10-06.tar.gz](https://github.com/vmware/harbor/files/513187/2016-10-06.tar.gz)

**Comment by hainingzhang** [comment]
Have you started up Harbor successfully before you enabled Swagger? It looks like you didn't run the ./prepare script.

**Comment by vikas027** [comment]
Hello @hainingzhang Yes, my Harbor was running well for `box88 (192.168.1.88)` and I did ran the `prepare-swagger.sh` script.

**Comment by hainingzhang** [comment]
Hi @vikas027  , 

What I meant was that you should install and run Harbor correctly first. From the logs, Harbor has not been correctly configured. Please read the below installation and configuration guide here first:

[https://github.com/vmware/harbor/blob/master/docs/installation_guide.md](https://github.com/vmware/harbor/blob/master/docs/installation_guide.md)

After you successfully install and run Harbor, you can then follow the Swagger configure guide to set up Harbor for Swagger.

**Comment by vikas027** [comment]
@hainingzhang I already have a running Harbor on the other server http://192.168.1.88 which I had installed through the tar ball `harbor-offline-installer-0.4.1.tgz`.  I can pull/push repos to/from it, thus I am assuming it is good.

I think Swagger and Harbor needs to be running at the same server. These are the complete steps which worked for me, hopefully will help someone else.

```
## For Harbor
root@box87:~# git clone https://github.com/vmware/harbor.git
root@box87:~# cd harbor/Deploy
root@box87:~/harbor/Deploy# sed -i 's/reg.mydomain.com/192.168.1.87/' harbor.cfg
root@box87:~/harbor/Deploy# ./prepare

## For Swagger
root@box87:~/harbor/Deploy# cd ../docs/
root@box87:~/harbor/docs# sed -i 's/reg.mydomain.com/192.168.1.87/' prepare-swagger.sh
root@box87:~/harbor/docs# ./prepare-swagger.sh

root@box87:~/harbor/docs# cd ../Deploy/
root@box87:~/harbor/Deploy# vi docker-compose.yml (added last two lines)
  ui:
    build:
      context: ../
      dockerfile: Deploy/ui/Dockerfile
    env_file:
      - ./config/ui/env
    restart: always
    volumes:
      - ./config/ui/app.conf:/etc/ui/app.conf
      - ./config/ui/private_key.pem:/etc/ui/private_key.pem
      - ../static/vendors/swagger-ui-2.1.4/dist:/go/bin/static/vendors/swagger
      - ../static/resources/yaml/swagger.yaml:/go/bin/static/resources/yaml/swagger.yaml

root@box87:~/harbor/Deploy# docker-compose build
root@box87:~/harbor/Deploy# docker-compose up -d
```

Now, `Harbor` is available at http://192.168.1.87
and `Swagger` is available at http://192.168.1.87/static/vendors/swagger/index.html

**Comment by hainingzhang** [comment]
Thanks for the update. 

Yes, Harbor and Swagger must run on the same server. We will update the document for this.
